onnxruntime/test/providers/cpu/tensor/pad_test.cc

```diff
@@ -809,22 +809,74 @@ TEST(PadOpTest, ConstantPadAxes) {
   OpTester test("Pad", 18);
   test.AddAttribute("mode", "constant");
   test.AddInput<int32_t>("data", {1, 2, 2, 2},
-  {
-    1, 1,
-    1, 1,
-    1, 1,
-    1, 1});
+                         {1, 1,
+                          1, 1,
+                          1, 1,
+                          1, 1});
   test.AddInput<int64_t>("pads", {4}, {0, 1, 0, 1});
   test.AddInput<int32_t>("value", {1}, {0});
+#if defined(USE_COREML) && defined(__APPLE__)
+  test.AddInput<int64_t>("axes", {2}, {1, 3}, true /*axes_is_initializer*/);
+#else
   test.AddInput<int32_t>("axes", {2}, {1, 3});
+#endif
   test.AddOutput<int32_t>("output", {1, 2, 2, 4},
-  {
-    0, 1, 1, 0,
-    0, 1, 1, 0,
-    0, 1, 1, 0,
-    0, 1, 1, 0
-    }
-  );
+                          {0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0});
+  test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
+}
+
+TEST(PadOpTest, ConstantPadAxesWithLast2Dim) {
+  OpTester test("Pad", 18);
+  test.AddAttribute("mode", "constant");
+  test.AddInput<int32_t>("data", {1, 2, 2, 2},
+                         {1, 1,
+                          1, 1,
+                          1, 1,
+                          1, 1});
+  test.AddInput<int64_t>("pads", {4}, {0, 1, 0, 1});
+  test.AddInput<int32_t>("value", {1}, {0});
+#if defined(USE_COREML) && defined(__APPLE__)
+  // input test case is supported by CoreML EP
+  test.AddInput<int64_t>("axes", {2}, {2, 3}, true /*axes_is_initializer*/);
+#else
+  test.AddInput<int32_t>("axes", {2}, {2, 3});
+#endif
+  test.AddOutput<int32_t>("output", {1, 2, 2, 4},
+                          {0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0});
+  test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
+}
+
+TEST(PadOpTest, ConstantPadAxesWithLast2Dim_2) {
```

Comment:
The intent of this test vs the previous isn't really clear from the naming. A more verbose test name or a comment explaining the intent of each test would be very helpful. 

Seems like the previous is validating that zero pads for an explicitly specified axis is handled correctly. Not sure what value the second test is adding given we successfully padded a dimension listed in axes in the previous test. Is there some edge case this hits that the previous test didn't?

FWIW the 'Last2Dim' seems somewhat inconsequential beyond that's a requirement for coreml to be able to potentially take the node, so I'd be inclined for that to be a comment on the line where we specify the axes rather than part of the test name.

Reply:
sure make sense. it would be clearer if we add a comment for explaining the test case instead of a vague test case name.

as for "this test vs. the previous", it's just for the purpose of adding more test cases for `specified axes` as suggested. but I agree the first one already tested we can successfully pad one dimension listed in axes.